core/lls_core/models/deskew.py

```diff
@@ -192,6 +192,15 @@ def read_image(cls, values: dict):
         # If the image was convertible to AICSImage, we should use the metadata from there
         if aics:
             values["input_image"] = aics.xarray_dask_data
+            
+            #check if final frame is complte. If not, get rid of it
+            final_frame = values["input_image"][-1]
+            try:
+                final_frame.compute()
+            except ValueError as e:
+                print("Final frame is borked. Acquisition probably stopped prematurely. Removing final frame.")
+                values["input_image"] = values["input_image"][0:-1]
+ 
```

Comment:
This would be neater as a "post" validator for the image, because then you can work with the `DataArray` directly.

Reply:
I didn't say it was neat, just that it worked :D 

I'm not currently really across pydantic and couldn't find where the data was first accessible within the model, hence putting it here. Happy to take some guidance though

Reply:
You can add a new function to the class such as:
```python

    @validator('input_image')
    def incomplete_acquisition(cls, v: DataArray):
        # Do something with v
```

This is a standard Pydantic validator, as explained here: https://docs.pydantic.dev/1.10/usage/validators/.

Reply:
Awesome, thanks I'll give it a shot and update the pull request if I can get it working as expected

Reply:
Thanks @multimeric - I've moved it into a validator as suggested.

Reply:
Perfect, that looks good.